### PR TITLE
Update config/Kconfig to correct MDK MuC Version

### PR DESCRIPTION
**Pull request by Kmart777**
Update config/Kconfig to correct MDK MuC version from STM32L476JG to STM32L476MG and so that menuconfig Board Selection Attributes point to desired user configuration options.

### CODE REVIEW

**Comment by Kmart777** [comment]
The push for arch: arm: Kconfig: Arrange HW selections to top of menuconfig GUI is optional.  I created 3rd party training material mirrored at https://drive.google.com/drive/folders/0B-w9A7ZbX8s4M3NNdzMySC13SXM that associates menuconfig attributes with their corresponding Kconfig files and line numbers.  Louis Begin is providing the .c and .h associations for each of the 3998 CONFIG_ variables.   I rearranged this section of code because I felt it would make a more intuitive user experience if HW selections were placed at the top of menuconfig.

**Review by flowflow25 — CHANGES_REQUESTED** [review]
As I recall we did NOT want to reorder this as it is intentionally in alphabetical order.

File: nuttx/arch/arm/src/stm32/Kconfig [hunk 1]
```diff
@@ -1690,7 +1689,184 @@ config STM32_ADC_INIT
...
+	bool "ADC1 Channel 14 (IN15-PB0)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN15_INIT
+	bool "ADC1 Channel 15 (IN16-PB1)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+endmenu
+endif
+endif
```
Comment (flowflow25) [inline nuttx/arch/arm/src/stm32/Kconfig]: Where's the else?  This broke the use of STM32_ADC for anyone NOT using STM32_STM32L4

**Review by flowflow25 — COMMENTED** [review]
Did you build a firmware or two with these and make sure they still worked?

are you going to do another change to update all the sample projects to the correct chip?

File: nuttx/arch/arm/src/stm32/Kconfig (hunk 1 above)
Comment (Kmart777) [reply nuttx/arch/arm/src/stm32/Kconfig]: I can't find a Kconfig example were ELSE is used, just multiple groupings of IF to ENDIF

**Review by flowflow25 — APPROVED** [review]

**Comment by Kmart777** [comment]
Replaced after consultation with flowflow25

**Pull request closed without merging**